Bitly/Bitly_Retrieve_Bitlink.ipynb

```diff
@@ -0,0 +1,253 @@
+   "id": "d7cb8a4e-fc0a-4707-bc72-9c16ed06c9ed",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "**Description:** This notebook will return information for a specified bitlink using the Bitly API."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd87de5b-8a93-470f-80da-0fab6eda9854",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "<u>References:</u>\n",
+    "- [Bitly API Documentation](https://dev.bitly.com/v4_documentation.html)\n",
+    "- [Bitly API Quickstart](https://dev.bitly.com/v4/#section/Quick-Start)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "21fe165c-8dea-419d-a99f-427f71f594c7",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "## Input"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6cd85c67-0488-45bb-b50c-70018ad3590c",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "### Import libraries"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "912afa92-2391-44be-be30-d3d52262ed5a",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import requests\n",
+    "import json\n",
+    "import naas\n",
+    "from pprint import pprint"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7dd928ae-e73d-4031-adc6-c1d0bdf51dff",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "### Setup Variables\n",
+    "- **token**: [Generate a Bitly Access Token](https://support.bitly.com/hc/en-us/articles/230647907-How-do-I-generate-an-OAuth-access-token-for-the-Bitly-API-)\n",
+    "- **bitlink**: A Bitlink made of the domain and hash "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "45dfd5f9-de84-4d78-be84-58639a9e45d9",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "token = naas.secret.get(\"BITLY_TOKEN\") or \"<YOUR_TOKEN_HERE>\"\n",
+    "bitlink = \"bit.ly/3lU6xxxt\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bfdfbe5e-2d07-42c4-bdca-ee55b432f069",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "## Model"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f01ad839-7e6d-4230-b72e-9308be5d1b5d",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "source": [
+    "### Retrieve a Bitlink"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70525d70-aa86-425b-b41d-9d495fa44c17",
+   "metadata": {
+    "papermill": {},
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "def retrieve_bitlink(token, bitlink):\n",
+    "    headers = {\"Authorization\": f\"Bearer {token}\"}\n",
+    "    url = f\"https://api-ssl.bitly.com/v4/bitlinks/{bitlink}\"\n",
+    "    response = requests.get(url, headers=headers)\n",
+    "    response.raise_for_status()\n",
```

Comment:
@FlorentLvr Getting an HTTPError here.

<img width="1337" alt="Screenshot 2023-02-24 at 7 04 33 PM" src="https://user-images.githubusercontent.com/81156510/221191806-719ea149-bb66-49ca-afa1-6095e89898d8.png">

Reply:
@srini047, it seems you are using the link in demo that why it is not working. You have to use a bitlink from your account. :)